mne/annotations.py

```diff
@@ -430,28 +430,78 @@ def _write_annotations(fid, annotations):
     end_block(fid, FIFF.FIFFB_MNE_ANNOTATIONS)
 
 
-def read_annotations(fname):
-    """Read annotations from a FIF file.
+def read_annotations(fname, sfreq='auto', uint16_codec=None, orig_time=None):
+    """Read annotations from a file.
+
+    This function reads a .fif, .fif.gz, .vrmk, edf or .set file and makes an
+    :class:`mne.Annotations` object.
 
     Parameters
     ----------
     fname : str
         The filename.
 
+    sfreq : float | 'auto'
+        The sampling frequency in the file. This parameter is necessary for
+        vmrk files as Annotations are expressed in seconds and vmrk files are
+        in samples. If set to 'auto' then the sfreq is taken from the .vhdr
+        file that has the same name (without file extension). So data.vrmk
+        looks for sfreq in data.vhdr.
+
+    uint16_codec : str | None
+        If your \*.set file contains non-ascii characters, sometimes reading
+        it may fail and give rise to error message stating that "buffer is
+        too small". ``uint16_codec`` allows to specify what codec (for example:
+        'latin1' or 'utf-8') should be used when reading character arrays and
+        can therefore help you solve this problem.
+
+    orig_time : float | int | instance of datetime | array of int | None
+        A POSIX Timestamp, datetime or an array containing the timestamp as the
+        first element and microseconds as the second element. Determines the
+        starting time of annotation acquisition. If None (default),
+        starting time is determined from beginning of raw data acquisition.
+        In general, ``raw.info['meas_date']`` (or None) can be used for syncing
+        the annotations with raw data if their acquisiton is started at the
+        same time.
+
     Returns
     -------
     annot : instance of Annotations | None
         The annotations.
     """
-    ff, tree, _ = fiff_open(fname, preload=False)
-    with ff as fid:
-        annotations = _read_annotations(fid, tree)
+    # XXX: I've issues with circular imports
+    from mne.io.brainvision.brainvision import _read_annotations_brainvision_caller
```

Comment:
I would remove the _caller

Reply:
I'm not sure. What I think it should be done is unify them all.
I would propose to remove all the `Annotation` construction of the readers and delegate it here to `mne.annotations.read_annotations`. In this manner, all readers should have a function `_read_annotations_XX`  (or parse or whatever) which does know about the file format and returns valid `onsets`, `duration`, `description` and `orig_time`. And remove the import Annotations.

However, this would be easier when `read_annotations_eeglab` is no longer there, and the synthesis of the stim channel is removed from the edf reader.

Reply:
but meanwhile, I'll just remove the callers.